pkg/sessionctx/stmtctx/stmtctx.go

```diff
@@ -468,7 +511,26 @@ func NewStmtCtxWithTimeZone(tz *time.Location) *StatementContext {
 }
 
 // Reset resets a statement context
-func (sc *StatementContext) Reset() {
+func (sc *StatementContext) Reset() bool {
+	// make sure no other goroutines still get locks.
+	if sc.mu != nil {
+		if !sc.mu.TryLock() {
+			return false
```

Comment:
In this case, it always means a bug: someone is still referencing this statement context, but we try to reset.

If we could solve why that happen, that's much better.  But it's too difficult to figure out why.
So this is a workaround to get rid of the fatal error ... panic.

Reply:
We can know that this situation has occurred in a certain place, but it is difficult to avoid all situations. `TryLock` here could avoid panic again in any other situations.